core/src/gl/error.h

```diff
@@ -18,10 +18,18 @@ class Error {
      */
     static bool hadGlError(const std::string& _locationTag);
 
+    static void glError(const char* stmt, const char* fname, int line);
+
 private:
 
     static std::unordered_map<GLenum, std::string> s_GlErrorCodesToStrings;
 
 };
 
+#ifdef TANGRAM_DEBUG
+#define GL_CHECK(STMT) do { STMT; Tangram::Error::glError(#STMT, __FILE__, __LINE__); } while (0)
```

Comment:
We don't have to wrap `STMT` in a set of parenthesis to be able to write `GL_CHECK()`, the macro would be an extra `;` in that case. 
I'm not sure we want something that clever, there's a pretty light set of gl functions we use that return a value, we already have all of the information we need to debug with this.